src/Raven.Server/Web/System/BackupServerWideHandler.cs

```diff
@@ -0,0 +1,230 @@
+                    backupsResult.Results.Add(backup);
+                }
+                
+                context.Write(writer, backupsResult.ToJson());
+                writer.Flush();
+
+                return Task.CompletedTask;
+            }
+        }
+        
+        // Toggle Disable/Enable for specified task
+        [RavenAction("/admin/configuration/serve-wide/backup/state", "POST", AuthorizationStatus.ClusterAdmin)]
+        public async Task ToggleTaskState()
+        {
+            var disable = GetBoolValueQueryString("disable") ?? true;
+            var taskName = GetStringQueryString("taskName", required: false);
+           
+            using (ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+            using (context.OpenReadTransaction())
+            {
+                // Get existing task
+                var backups = ServerStore.Cluster.GetServerWideBackupConfigurations(context, taskName); 
+                ServerWideBackupConfigurationResults backupsResult = new ServerWideBackupConfigurationResults();
+                
+                if (backups.Count() == 0)
+                    throw new InvalidOperationException($"Server-Wide Backup Task: {taskName} was not found in the server.");
+                
+                var serveWideBackupBlittable = backups.ToList()[0];
+                ServerWideBackupConfiguration serverWideBackup = JsonDeserializationServer.ServerWideBackupConfiguration(serveWideBackupBlittable);
+                
+                // Toggle
+                serverWideBackup.Disabled = disable; 
+            
+                // Save task
+                var (newIndex, _) = await ServerStore.PutServerWideBackupConfigurationAsync(serverWideBackup, GetRaftRequestIdFromQuery());
+                await ServerStore.WaitForCommitIndexChange(RachisConsensus.CommitIndexModification.GreaterOrEqual, newIndex);
+
+                using (var writer = new BlittableJsonTextWriter(context, ResponseBodyStream()))
+                {
+                    var test = new ModifyServerWideBackupResult() 
+                    {
+                        TaskName = taskName,
+                        RaftCommandIndex = newIndex
+                    };
+
+                    HttpContext.Response.StatusCode = (int)HttpStatusCode.OK;
+                    context.Write(writer, test.ToJson());
+                    writer.Flush();
+                }
+            }
+        }
+        
+        [RavenAction("/admin/backup-data-directory", "GET", AuthorizationStatus.ClusterAdmin)]
+        public async Task FullBackupDataDirectory()
+        {
+            var path = GetStringQueryString("path", required: true);
+            var requestTimeoutInMs = GetIntValueQueryString("requestTimeoutInMs", required: false) ?? 5 * 1000;
+
+            var pathResult = BackupConfigurationHelper.GetActualFullPath(ServerStore, path);
```

Comment:
Create a static method that will take path, requestTimeoutInMs and getNodesInfo as a list of arguments and will perform the operation. Use that in both places so we will not have desync in the future between those two endpoints.

Reply:
done